File: 31.07.25 Т-Банк Т-академия/A.java
import java.util.Scanner;

public class A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        sc.close();

        if (n >= 8) {
            // Если n >= 8, просто вычитаем 7
            System.out.println(n - 7);
            return;
        }

        // Проверяем длину месяца, чтобы определить результат
        int monthLen = 28;
        if (!daysOnMonth(n, m, 28)) {
            if (daysOnMonth(n, m, 29)) {
                monthLen = 29;
            } else if (daysOnMonth(n, m, 30)) {
                monthLen = 30;
            } else {
                monthLen = 31;
            }
        }

        System.out.println(n - 7 + monthLen);
    }

    private static boolean daysOnMonth(int n, int m, int days) {
        if (m > days) {
            return false;
        }
        return (n + days - m) == 14;
    }
}

File: 31.07.25 Т-Банк Т-академия/A.py
def days_on_month(days: int):
    global n, m
    if m > days:
        return False
    return (n+days-m) == 14


n, m = map(int, input().strip().split())
if n >= 8:
    print(n-7)
    exit(0)
else:
    month_len = 0
    if days_on_month(28):
        month_len = 28
    elif days_on_month(29):
        month_len = 29
    elif days_on_month(30):
        month_len = 30
    else:
        month_len = 31
    print(n-7+month_len)
